### DISCUSSION TITLE
HR/Groups in Dropdown field

### DISCUSSION THREAD

**Discussion by gbdesign2023** [tobimori/kirby-dreamform#70] [Feature Requests]
**More options for dropdown-field:** So far, I like to use the first entry as an inactive label for dropdown-fields. This can also be used to remove a selection. This is not possible with a "classic" field once a selection has been made. Important, for example, for an optional entry to be cancelled. 
I also think it is a good idea to structure the selection list with an `<hr>` element. 

Proposal:
```
<select id="select01" name="select01">
<option disabled selected>Please select:</option>
<hr>
<option value="phpe">PHP</option>
<option value="css">CSS</option>
<option value="js">Javascript</option>
<hr>
<option value="markdown">Markdown</option>
<option value="yaml">YAML</option>
</select>
```

**Comment by tobimori** [comment]
- If you specify a placeholder for the select field, it's set as you wrote in your proposal.  
- Dividers/groups in selects is a harder one, since I'd need to change the field for storing options in the field from a structure field to a Blocks field, so people can chose which type to add. For now, I'd recommend using a custom field for that.
- Links in checkboxes is a duplicate of #57. I will therefore remove it from this FR.

**Comment by gbdesign2023** [comment]
When the dropdown was first clicked, the placeholder was visible. Once a value has been selected, the placeholder disappears. It is therefore no longer possible to reset the selection. 

- _If you specify a placeholder for the select field, it's set as you wrote in your proposal._

I had a closer look at the snippet and only a small change was required for my solution:

<img width="317" alt="Bildschirmfoto 2024-07-09 um 15 53 54" src="https://github.com/tobimori/kirby-dreamform/assets/140849227/f08d4867-4abf-4839-9732-c509af2bcc1b">


```
<select class="form-select form-select-lg rounded-0" <?= attr(A::merge($attr['input'], [
	'id' => $form->elementId($block->id()),
	'name' => $block->key(),
	'required' => $required ?? null,
])) ?>>
	<option <?= attr([
		"selected" => !($selected = $form->valueFor($block->key())?->value()),
		"value" => true,
		"disabled" => true,
		"hidden" => false // <---- set to false did the job
	]) ?>><?= $block->placeholder()->escape() ?></option>
	<hr />
	<?php foreach ($field->options() as $value => $label) : ?>
	...
```

- _Links in checkboxes is a duplicate of https://github.com/tobimori/kirby-dreamform/discussions/57. I will therefore remove it from this FR._

Thank you also for pointing out that there is already an enquiry about this.

**Reply by tobimori** [reply]
Do you think it would be a good solution to set `hidden` to `false` when the field is not set as `required`?

**Reply by gbdesign2023** [reply]
As long as the "hidden" tag is assigned to the placeholder, even if an entry has been made: yes
Shouldn't the "hidden" tag disappear as soon as a placeholder text has been entered?
Is there a mistake or misunderstanding on my part?

```
<select id="crNjV1neQBWtcljo/b8a83fa7-1cd0-447d-8a31-644d0ffeb186" name="anrede">
<option disabled hidden selected value>Bitte auswählen:</option>
<hr>
<option value="frau">Frau</option>
<option value="herr">Herr</option>
</select>
```

**Reply by tobimori** [reply]
I added hidden because I especially not wanted it to be visible in the dropdown

**Reply by gbdesign2023** [reply]
It is effortless to deactivate/activate this option in the template. Since I had to customise the template anyway because of my Bootstrap framework, this is not a disadvantage. But now I have found a solution, and thank you for your support.
